sensor/kubernetes/fake/deployment.go

```diff
@@ -6,18 +6,59 @@ import (
 	"math/rand"
 	"time"
 
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/containerid"
 	"github.com/stackrox/rox/pkg/fixtures"
 	"github.com/stackrox/rox/pkg/k8sutil"
 	"github.com/stackrox/rox/pkg/kubernetes"
 	"github.com/stackrox/rox/pkg/pointers"
+	"github.com/stackrox/rox/pkg/sync"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var (
+	processPool = newProcessPool()
+)
+
+type ProcessPool struct {
+	ProcessPool	map[string][]*storage.ProcessSignal
+	lock		sync.RWMutex
+}
+
+func newProcessPool() *ProcessPool {
+	return &ProcessPool{
+		ProcessPool:	make(map[string][]*storage.ProcessSignal),
+	}
+}
+
+func (p *ProcessPool) add(val *storage.ProcessSignal) {
```

Comment:
Locks might not be needed in this function and the remove function, as I don't think there will be multiple simultaneous calls with the same containerId.